internal/parse/parser.go

```diff
@@ -142,13 +144,38 @@ func (p *Parser) Parse(input string) (expr *ParsedExpr, err error) {
 		}
 
 		// If nothing above can be parsed we advance the parser.
-		p.pos++
+		p.advance()
 	}
 	// Add any remaining unparsed string input to the parser.
 	p.add(nil)
 	return &ParsedExpr{p.parts}, nil
 }
 
+// advance increments p.pos until we reach a space, a $, a " or a '. If we find
+// a space we advance to the rightmost adjacent space so the p.pos points to the
+// last space before the next non-space character.
+func (p *Parser) advance() {
+	noteableBytes := map[byte]bool{
+		'$':  true,
+		'"':  true,
+		'\'': true,
+		' ':  true,
+	}
+	// Advance the parser to the next char.
```

Comment:
Same.. that's literally what `p.pos++` means. The English version of that is much longer. :-)

I'd argue the same about the remaining ones, but please feel free to decide whether to keep or not.

Reply:
Changed.